UNITTESTS/mbed_unittest.py

```diff
@@ -1,163 +1,14 @@
-
-    # Prepare build directory
-    tool.prepare_build_directory(path_to_src=pwd,
-                                 build_path=options.build,
-                                 clean=options.clean)
-
-    if options.compile_only:
-        # Create makefiles
-        src_path = os.path.relpath(pwd, options.build)
-        tool.create_makefiles(path_to_src=src_path,
-                              generator=options.cmake_generator,
-                              coverage_output_type=options.coverage,
-                              debug=options.debug_build,
-                              valgrind=options.valgrind)
-
-        # Build tests
-        tool.build_tests()
-
-    if options.run_only:
-        tool.run_tests(filter_regex=options.test_regex,
-                       valgrind=options.valgrind)
-
-        # If code coverage generation:
-        if options.coverage:
-            cov_api = CoverageAPI(
-                mbed_os_root=os.path.normpath(os.path.join(pwd, "..")),
-                build_dir=options.build)
-
-            # Generate reports
-            outputs = [options.coverage]
-            if options.coverage == "both":
-                outputs = ["html", "xml"]
-
-            excludes = [pwd, options.build]
-
-            if not options.include_headers:
-                excludes.append(re.compile(".*\\.h"))
-
-            cov_api.generate_reports(outputs=outputs,
-                                     excludes=excludes,
-                                     build_path=options.build)
-
-def _mbed_unittest_new(options, pwd):
-    if options is None:
-        return
-
-    generator = UnitTestGeneratorTool()
-
-    mbed_os_root = os.path.normpath(
-        os.path.join(pwd, "..")
-    )
-
-    for filepath in options.new_files:
-        generator.create(
-            mbed_os_root=mbed_os_root,
-            filepath=filepath)
-
-def mbed_unittest(options=None):
-    """
-    Create, build and run unit tests
-    """
-
-    if options is None:
-        return
-
-    cwd = os.getcwd()
-
-    # Change current working directory to script location
-    unittest_dir = os.path.dirname(os.path.realpath(__file__))
-    os.chdir(unittest_dir)
-
-    if options.new_files:
-        _mbed_unittest_new(options, unittest_dir)
-    else:
-        _mbed_unittest_test(options, cwd, unittest_dir)
-
-def mbed_unittest_main():
-    """
-    Create, build and run unit tests - main
-    """
-
-    # Get parser and parse command line arguments
-    parser = get_options_parser()
-    options = parser.parse_args()
-
-    # Setup logger
-    if options.log_level:
-        logging.basicConfig(level=getattr(logging, options.log_level),
-                            format="%(message)s")
-    else:
-        logging.basicConfig(level=logging.INFO,
-                            format="%(message)s")
-
-    mbed_unittest(options)
 
 if __name__ == "__main__":
-    mbed_unittest_main()
+    os.system("cmake -DUNITTESTS=ON -S . -B cmake_build -GNinja")
```

Comment:
this currently doesn't do anything fancy but if we had some params passed in from `mbed test` we could run tests by name or label

Reply:
I would remove this python script completely, let a user to invoke the cmake and pass what it needs. Better  to learn how to build unittests and ctests .

Reply:
You can do that, sure, but if someone already has `mbed test` in CI or something this will let them continue using it without us having to continue to maintain the old build system

Reply:
> but if someone already has mbed test in CI or something this will let them continue using it without us having to continue to maintain the old build system

True, but with all the changes, something else might break. Better finding out now and fixing the CI to be compatible with the future (and enjoy the possibilities brought by the new build system) than having to maintain something so that users think they are still using the old tools.